compiler/rustc_const_eval/src/const_eval/eval_queries.rs

```diff
@@ -335,6 +335,44 @@ pub fn eval_to_allocation_raw_provider<'tcx>(
     tcx: TyCtxt<'tcx>,
     key: ty::PseudoCanonicalInput<'tcx, GlobalId<'tcx>>,
 ) -> ::rustc_middle::mir::interpret::EvalToAllocationRawResult<'tcx> {
+    // Avoid evaluating instances with impossible bounds required to hold as
```

Comment:
write down justification for why the check is here and not, say, in match handling or type system normalization etc